src/Popover/modifiers.js

```diff
@@ -11,35 +12,47 @@ const computeArrowPadding = () => {
     return Math.ceil(padding)
 }
 
-export const offset = {
-    name: 'offset',
-    options: {
-        offset: [0, ARROW_SIZE],
-    },
-}
+const hideArrowWhenDisplaced = ({ state }) => {
+    const halfArrow = ARROW_SIZE / 2
+    const displacement = state.modifiersData.preventOverflow
+    const referenceRect = state.rects.reference
+    const shouldHideArrow =
+        Math.abs(displacement.x) >= referenceRect.width + halfArrow ||
+        Math.abs(displacement.y) >= referenceRect.height + halfArrow
+
+    state.attributes.arrow = { ['data-arrow-hidden']: shouldHideArrow }
```

Comment:
Is there a reason this is passed as a data- prop under `arrow` rather than just as a normal `hidden` prop to the `Arrow` component?

Reply:
Seems like maybe I'm missing something about the `react-popper` API here?

Reply:
Yeah this is the way of popper.js, you add behaviour via modifiers. And, when used directly, popper.js will attach data-attributes to elements, hence the kebab-cased property names. We're mainly using modifiers from popper.js itself, which are already using that kebab-cased approach, so I decided to just stick to that in our custom modifiers.